### PR TITLE
[Buttons] Deprecating MDCContainedButtonColorThemer

### PR DESCRIPTION
**Pull request by galiak11**
## Description

Deprecate symbol MDCContainedButtonColorThemer(ToBeDeprecated)::applySemanticColorScheme:toButton:

## Issue

b/145204991 Delete symbol "MDCContainedButtonColorThemer(ToBeDeprecated)::applySemanticColorScheme:toButton:"

### CODE REVIEW

File: components/Buttons/src/Theming/MDCButton+MaterialTheming.m [hunk 1]
```diff
@@ -40,7 +40,19 @@ - (void)applyContainedThemeWithScheme:(nonnull id<MDCContainerScheming>)scheme {
 }
 
 - (void)applyContainedThemeWithColorScheme:(id<MDCColorScheming>)colorScheme {
-  [MDCContainedButtonColorThemer applySemanticColorScheme:colorScheme toButton:self];
+  [self resetButtonColorsForAllStates];
```
Comment (bryanoltman) [inline components/Buttons/src/Theming/MDCButton+MaterialTheming.m:43]: The change to this file seems unrelated to deprecating MDCContainedButtonColorThemer. Please move this to a separate PR.

File: components/Buttons/src/ColorThemer/MDCContainedButtonColorThemer.h [hunk 2]
```diff
@@ -27,7 +27,8 @@
 @interface MDCContainedButtonColorThemer : NSObject
 @end
 
-@interface MDCContainedButtonColorThemer (ToBeDeprecated)
+__deprecated_msg("Please use [MDCButton applyContainedThemeWithScheme:] instead. (Note: Color theming is no longer available as an independent API.)")
+    @interface MDCContainedButtonColorThemer (ToBeDeprecated)
```
Comment (bryanoltman) [inline components/Buttons/src/ColorThemer/MDCContainedButtonColorThemer.h]: Deprecating the category doesn't cause warnings to be generated by usages of `MDCContainedButtonColorThemer`. This PR should remove the `@interface MDCContainedButtonColorThemer (ToBeDeprecated)` category declaration and apply the deprecation warning directly to the @interface declaration above on line 25.

**Review by bryanoltman — DISMISSED** [review]

File: components/Buttons/src/Theming/MDCButton+MaterialTheming.m (hunk 1 above)
Comment (galiak11) [reply components/Buttons/src/Theming/MDCButton+MaterialTheming.m:43]: It is related because the deprecated API was used here and now it is not.

Comment (galiak11) [reply components/Buttons/src/Theming/MDCButton+MaterialTheming.m]: Done

File: components/Buttons/src/ColorThemer/MDCContainedButtonColorThemer.h (hunk 2 above)
Comment (galiak11) [reply components/Buttons/src/ColorThemer/MDCContainedButtonColorThemer.h]: Done

File: components/Buttons/src/Theming/MDCButton+MaterialTheming.m (hunk 1 above)
Comment (bryanoltman) [reply components/Buttons/src/Theming/MDCButton+MaterialTheming.m:43]: Please do not change Active libraries as part of this change – this PR should only be adding deprecation warnings. These changes should be a part a future PR to remove the class being deprecated in this PR.

**Review by bryanoltman — DISMISSED** [review]
Please do not dismiss a request for changes until the conversation is resolved

File: components/Buttons/src/Theming/MDCButton+MaterialTheming.m (hunk 1 above)
Comment (galiak11) [reply components/Buttons/src/Theming/MDCButton+MaterialTheming.m:43]: As discussed, future PRs will be separated.

**Review by bryanoltman — APPROVED** [review]
